Sources/HaCWebsiteLib/Workshop.swift

```diff
@@ -0,0 +1,103 @@
+import Foundation
+import Yaml
+import KituraMarkdown
+
+public struct Workshop {
+  let title: String
+  let authors: [String]
+  let presenters: [String]
+  let thanks: [Thanks]
+  let recommendations: [Recommendation]
+  let links: [URL]
+  let tags: [String]
+  let prerequisites: String
+  let description: String
+
+  struct Thanks: Equatable {
+    let to: String
+    let reason: String?
+  }
+
+  struct Recommendation: Equatable {
+    let title: String
+    let url: URL
+  }
+
+  enum WorkshopError : Error {
+    case malformedMetadata
+  }
+}
+
+// Satisfy equatability for Workshop.Recommendation
+func ==(lhs: Workshop.Recommendation, rhs: Workshop.Recommendation) -> Bool { 
+  return lhs.title == rhs.title && lhs.url == rhs.url
+}
+// ...and for Workshop.Thanks
+func ==(lhs: Workshop.Thanks, rhs: Workshop.Thanks) -> Bool {
+  return lhs.to == rhs.to && lhs.reason == rhs.reason
+}
+
+public extension Workshop {
+  /**
+    Parses the files that describe a workshop into a Workshop instance
+
+    - returns:
+      The generated Workshop instance
+   
+    - parameters:
+      - metadataYaml: The String contents of the Yaml metadata for this workshop
+      - descriptionMarkdown: The String contents of the Markdown file with the description of this workshop
+      - prerequisitesMarkdown: The String contents of the Markdown file with the prerequisites of this workshop
+   
+    - throws:
+      An error of type Workshop.WorkshopError.malformedMetadata if the Yaml file does not follow the expected structure
+   
+   */
+  public static func parse(metadataYaml: String, descriptionMarkdown: String, prerequisitesMarkdown: String) throws -> Workshop {
+    // Parse Yaml and Markdown
+    let metadata = try Yaml.load(metadataYaml)
+    let description = KituraMarkdown.render(from: descriptionMarkdown)
```

Comment:
Good point. We could store both, the idea of avoiding re-rendering the markdown for every serve is appealing to me, do you think it's too domain-specific to store at all?